react-app/src/components/Extract.js

```diff
@@ -0,0 +1,106 @@
+import { Col, Form, Row } from 'react-bootstrap'
+
+import '../stylesheets/Page.css';
+
+class Extract extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            configPath: "",
+            logPath: "",
+            includeDebug: false,
+            filterStart: false,
+            filterEnd: false
+        }
+
+        this.onConfigFileChange = this.onConfigFileChange.bind;
+        this.onLogFileChange = this.onLogFileChange.bind;
+        this.onIncludeDebugChange = this.onIncludeDebugChange.bind;
+        this.onFilterStartChange = this.onFilterStartChange.bind;
+        this.onFilterEndChange = this.onFilterEndChange.bind;
+        this.onSubmit = this.onSubmit.bind(this);
+    }
+
+    onConfigFileChange(e) {
+        this.setState({
+            configPath: e.target.value
+        });
+    }
+
+    onLogFileChange(e) {
+        this.setState({
+            logPath: e.target.value
+        });
+    }
+
+    onIncludeDebugChange(e) {
+        this.setState({
+            includeDebug: e.target.value
+        });
+    }
+
+    onFilterStartChange(e) {
+        this.setState({
+            filterStart: e.target.value
+        });
+    }
+
+    onFilterEndChange(e) {
+        this.setState({
+            filterEnd: e.target.value
+        });
+    }
+
+    onSubmit(e) {
+        console.log(e);
+        console.log(this.configPath);
+    }
+
+    render() {
+        this.whatever = 10;
+        return <div>
+            <h1 className="page-title">Extract New</h1>
+            <Form>
+                <Row>
+                    <Col>
+                        <Form.Group controlId="formConfigPath" className="mb-3">
+                            <Form.Label className="form-label" >Configuration File</Form.Label>
+                            <Form.Control type="text" />
+                        </Form.Group>
+                        <Form.Group controlId="formLogPath">
+                            <Form.Label className="form-label" >Previous Log File</Form.Label>
+                            <Form.Control type="text" />
+                        </Form.Group>
+                        <Form.Group controlId="formIncludeDebug">
+                            <Form.Check type="checkbox" label="Log output debugging information" />
+                        </Form.Group>
+                    </Col>
+                </Row>
+                <Row>
+                    <Col>
+                        <Form.Group controlId="formIncludeStartDate">
+                            <Form.Check type="checkbox" label="Filter by start date" />
+                        </Form.Group>
+                    </Col>
+                    <Col>
+                        <Form.Group controlId="formIncludeEndDate">
+                            <Form.Check type="checkbox" label="Filter by end date" />
+                        </Form.Group>
+                    </Col>
+                </Row>
+            </Form>
```

Comment:
I noticed that in the form elements here, we never use `value` or `onChange` to sync up what is typed/selected into the form with the actual state variables. Since we have the state variables and the functions defined, maybe we should link them up via the `value` and `onChange` props of these components so the users' input is properly saved.

Note that the console log for configPath for instance always logs empty string since the value is never updated via `setState`.